### DISCUSSION TITLE
Is a correct HTML document also a correct MDX document?

### DISCUSSION THREAD

**Discussion by techquestions** [mdx-js/mdx#1680] [General] [answered]
Let's say we have an MDX file with the following contents:

```html
<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 4.01//EN"
   "http://www.w3.org/TR/html4/strict.dtd">
<HTML>
   <HEAD>
      <TITLE>My first HTML document</TITLE>
   </HEAD>
   <BODY>
      <P>Hello world!
   </BODY>
</HTML>
```

Would that be a correct MDX document? If not, what is preventing it from it? 

I know that single MDX files are typically part of larger DOM structure, so embedding this in a JSX file and then actually rendering would result in an incorrect HTML document (that is, having `<html>` inside of `<body>` but let's assume this does not happen, for the sake of this question.

**Comment by ChristianMurphy** [comment, marked as answer]
> Is a correct HTML document also a correct MDX document?

No.

> Would that be a correct MDX document?

It would not be.

> If not, what is preventing it from it?

MDX the language is Markdown - HTML + JSX.
While JSX looks similar and sometimes identical to HTML.
JSX differs from HTML in a number of ways.
Including (but not limited to):
* JSX uses stricter XML style parsing
* JSX does not support doctype
* JSX is case sensitive, HTML is not
* JSX uses the `className` attribute to add CSS classes, HTML uses `class`

---

It sounds like you are interested in mixing markdown and HTML?
If so, good news, markdown itself supports this.
`remark` the underlying library which mdx extends can do this.
There's a guide on how to do this at https://unifiedjs.com/learn/recipe/remark-html/

In the future MDX the library may include a mode where it can be switched to parse plain markdown.
`xdm` https://github.com/wooorm/xdm which is currently merging back with mdx, has the capability to switch modes based on an option/file-extension.
